src/org/labkey/test/components/ui/entities/ParentEntityEditPanel.java

```diff
@@ -8,80 +8,114 @@
-    private final ParentType _parentType;
+    private final WebDriver driver;
+    private final WebElement editingDiv;
 
     /**
      * Constructor for the panel.
      *
      * @param element The WebElement containing the panel.
      * @param driver A reference to the WebDriver
-     * @param parentType An enum of type @ParentType used to identify the type of lineage entity is being edited.
-     *                   This is used to set the expected text when searching for web elements.
      */
-    public ParentEntityEditPanel(WebElement element, WebDriver driver, ParentType parentType)
+    public ParentEntityEditPanel(WebElement element, WebDriver driver)
     {
-        _driver = driver;
-        _editingDiv = element;
-        _parentType = parentType;
+        this.driver = driver;
+        editingDiv = element;
     }
 
     @Override
     public WebElement getComponentElement()
     {
-        return _editingDiv;
+        return editingDiv;
     }
 
     @Override
     protected WebDriver getDriver()
     {
-        return _driver;
+        return driver;
     }
 
-    // TODO: It would be nice if the save and cancel buttons would return a specific page type. That would require
-    //  using generics (much in the same way the modal dialog does), but I don't have time to do that right now.
-
-    /**
-     * Check to see if the panel is loaded. A panel is considered loaded if all of the combo-boxes in the panel are
-     * valid and are not loading.
-     *
-     * @return True if all of the combo-box references are valid and not loading.
-     */
-    public boolean isPanelLoaded()
+    @Override
+    protected void waitForReady()
     {
-        return getAllTypeCombo().stream().allMatch(rs -> rs.isInteractive() && !rs.isLoading());
+
+        // The panel is ready if:
+        // 1. There are no spinners.
+        // 2. If there are currently no parents there should be only one entity type select (ReactSelect) and it should be ready.
+        // 3. If there are parents (i.e. FilterReactSelect controls present) none of them should be empty.
+        WebDriverWrapper.waitFor(()->
+        {
+            if(BootstrapLocators.loadingSpinner.findWhenNeeded(this).isDisplayed())
+            {
+                return false;
+            }
+            else
+            {
+
+                // Check to see if there are no existing parents.
+                List<ReactSelect> firstParentSelector = ReactSelect.finder(getDriver())
+                        .withNamedInput("entityType0")
+                        .findAll(this);
+
+                Assert.assertFalse("It looks like there are multiple new parent ReactSelect controls present, that should never happen.",
+                        firstParentSelector.size() > 1);
+
+                if(!firstParentSelector.isEmpty())
+                {
+                    return firstParentSelector.get(0).isInteractive();
+                }
```

Comment:
I think you want to check the state of the entityType selector second. `firstParentSelector` should never never be empty, so this will always return before checking for any selected source/parent IDs.

Reply:
Ok, hopefully third time is the charm.